sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaSuite.scala

```diff
@@ -1390,7 +1395,11 @@ class ParquetSchemaSuite extends ParquetSchemaTest {
     catalystSchema = new StructType(),
 
     expectedSchema = ParquetSchemaConverter.EMPTY_MESSAGE,
-    caseSensitive = true)
+    conf = {
+      val conf = new Configuration()
+      conf.setBoolean(SQLConf.CASE_SENSITIVE.key, true)
```

Comment:
isn't it the default value?

Reply:
@cloud-fan There is no default value for `spark.sql.caseSensitive` in Configuration. Let me explain in more details below.

This is one of the overloaded methods of `testSchemaClipping`. I tried to give this `testSchemaClipping` method a default conf, however Scalac complains that 
> in class ParquetSchemaSuite, multiple overloaded alternatives of testSchemaClipping define default arguments

https://github.com/apache/spark/blob/95673cdacb1da65d7ac45c212a365e167ae0d713/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaSuite.scala#L1014-L1019

https://github.com/apache/spark/blob/95673cdacb1da65d7ac45c212a365e167ae0d713/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaSuite.scala#L1028-L1033

It seems a little confusing, because these two methods have different parameter types. After a brief investigation, I found Scala compiler simply disallows overloaded methods with default arguments even when these methods have different parameter types.

https://stackoverflow.com/questions/4652095/why-does-the-scala-compiler-disallow-overloaded-methods-with-default-arguments